local.c

```diff
@@ -1488,6 +1494,22 @@ static int detect_and_move_global_attrs(struct iio_device *dev)
 	return 0;
 }
 
+static int add_buffer_attr(void *d, const char *path)
+{
+	struct iio_device *dev = (struct iio_device *) d;
+	char buf[1024];
+	const char *name = strrchr(path, '/') + 1;
+	int i;	
+
+	for (i = 0; i < ARRAY_SIZE(buffer_attrs_reserved); i++)
+		if (!strcmp(buffer_attrs_reserved[i], name))
+			return 0;
+
+	iio_snprintf(buf, sizeof(buf), "buffer/%s", name);
+
+	return add_attr_to_device(dev, buf);
```

Comment:
How do we avoid clashes if the buffer and the device have attributes with the same name?

We initially added the scan_element attributes to the channel attributes, but then eventually clashes happened and we had to undo that. I'd like to avoid that for the buffer attributes/

Reply:
The "buffer/" is retained in the name. So you would have attributes named "buffer/data_available"; this should disambiguate between any other attribute names

Reply:
For example:

```
C:\Work\libiio\build\examples\Debug>iio_info -n 10.0.0.220
Library version: 0.11 (git tag: cb2f40f)
Compiled with backends: xml ip usb serial
IIO context created with network backend.
Backend version: 0.11 (git tag: v0.11  )
Backend description string: 10.0.0.220 Linux buildroot-x86mw 4.9.0-00622-g43fb8a6-dirty #21 Tue Nov 14 20:17:29 EST 2017 x86_64
IIO context has 2 attributes:
        local,kernel: 4.9.0-00622-g43fb8a6-dirty
        ip,ip-addr: 10.0.0.220
IIO context has 2 devices:
        iio:device1: mwipcore0:mm2s0 (buffer capable)
                1 channels found:
                        data0:  (output, index: 0, format: le:U32/32>>0)
                3 device-specific attributes found:
                                attr  0: reset_ip_mode value: none
                                attr  1: reset_ip_mode_available value: none enable disable all
                                attr  2: buffer/data_available value: 0
```

Reply:
@lclausen-adi -- does the use of a "buffer/" namespace resolve your concern about collisions with other attribute names, or is there some other issue I'm missing?